src/esp/gfx/replay/Keyframe.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include "esp/assets/Asset.h"
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+
+namespace esp {
+namespace scene {
+class SceneNode;
+}
+namespace gfx {
+namespace replay {
+
+using RenderAssetInstanceKey = uint32_t;
+
+// serialization/replay-friendly Transform class
+struct Transform {
```

Comment:
I don't use Transform and RenderAssetInstanceState in this PR, but they do showcase the serialization code. If you have questions/concerns about what exactly I'm serializing for the render-replay feature, I suggest waiting for that PR (coming soon).